File: DSAlgorithms/Leetcode/binary_search/assorted/c++/977-sqs_of_sorted_arr.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
       std::vector<int> result; 
       int left = 0;
       int right = nums.size() - 1;
        while(left <= right)
        {
            if(nums[left] * nums[left] > nums[right] * nums[right])
            {
                result.push_back(nums[left] * nums[left]);
                left++;
            }
            else
            {
                result.push_back(nums[right] * nums[right]);
                right--;
            }
        }
        reverse(result.begin(), result.end());
        return result;
    }
};

File: dsa_algo/leetcode/ts/src/binary_search/977-sqs_of_sorted_arr.ts
const sortedSquares = (nums: number[]): number[] =>{
    const result: number[] = [];
    let right = nums.length - 1;
    let left = 0
    while(left <= right)
    {
        if(nums[left] * nums[left] > nums[right] * nums[right])
        {
            result.push(nums[left] * nums[left]);
            left++;
        }
        else
        {
            result.push(nums[right] * nums[right]);
            right--;
        }
    }
    result.reverse();
    return result;
};